packages/gatsby/src/commands/build.ts

```diff
@@ -232,14 +238,34 @@ module.exports = async function build(program: IBuildArgs): Promise<void> {
   try {
     const result = await buildRenderer(program, Stage.BuildHTML, buildSpan)
     pageRenderer = result.rendererPath
+    if (_CFLAGS_.GATSBY_MAJOR === `4` && shouldGenerateEngines()) {
+      // for now copy page-render to `.cache` so page-ssr module can require it as a sibling module
+      const outputDir = path.join(program.directory, `.cache`, `page-ssr`)
```

Comment:
Can you rename it to _routes as it's more in par with https://github.com/gatsbyjs/gatsby/pull/33054
```suggestion
      const outputDir = path.join(program.directory, `.cache`, `_routes`)
```